### ISSUE TITLE
skhd: could not lock pid-file! abort..

### ISSUE THREAD

**Issue by aravezskinteeth** [koekeishiya/skhd#243]
# Description of the issue

After installing and making a config file, none of my hotkeys work. When trying any of the specified commands, like `skhd -V`, `skhd -P` or even trying to configure the path of my config path, the output is:

```shell
skhd: could not lock pid-file! abort..
```

# System Specifications

OS: MacOS Monterrey 12.6.2

# Steps taken to reproduce the problem

1. Installed `skhd`

```shell
  brew install koekeishiya/formulae/skhd
```

2. Created a config file following the documentation and other peoples' examples. Saved it on this path: `~/.config/skhd`

3. Started `skhd`

```shell
  brew services start skhd
```

# Things I have already tried before 

Looked up the issue on github, found this #74 #126 

I only have sketchybar, yabai and skhd as LaunchAgents, and looking at the `skhd plist` file, which looks like this:

```plist
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
  <key>Label</key>
  <string>homebrew.mxcl.skhd</string>
  <key>ProgramArguments</key>
  <array>
    <string>/usr/local/opt/skhd/bin/skhd</string>
  </array>
  <key>EnvironmentVariables</key>
  <dict>
    <key>PATH</key>
    <string>/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin</string>
  </dict>
  <key>RunAtLoad</key>
  <true/>
  <key>KeepAlive</key>
  <true/>
  <key>ProcessType</key>
  <string>Interactive</string>      
  <key>Nice</key>
  <integer>-20</integer>
</dict>
</plist>
```

So I wonder If this has to do with the Standard Error Path missing. Because, when I tried to modied it as per #126 , it still didn't work. So I tried to find if there was a `/tmp/skhd.err` file on my system and couldn't find any. 

# Note

By the way, I'm relatively new at how launch services work. I'm trying to research it on my own, but I'm having trouble and I really need help. I probably messed something up.

**Comment by nipunravisara** [comment]
How did you get this fixed. Could you elaborate more?

**Comment by aravezskinteeth** [comment]
> How did you get this fixed. Could you elaborate more?

Yes, I solved this permanently and I don't have to use `sudo` anymore. 

## Enable root password

1. Open Spotlight with <kbd>cmd</kbd> + <kbd>space</kbd>
2. Type `Directory Utility.app` 
3. Click the lock icon, then enter your admin account name and password.
4. Click on `Edit`--> `Enable Root User` in the menu bar. Then enter the password that to want to use and save it on a note. 

## Enable user permissions to the /private/tmp directory

1. Open Finder
2. Press <kbd>shift</kbd> + <kbd>cmd</kbd> + <kbd>G</kbd>
3. Type `/private/tmp/`
4. Right click on the `skhd_username.pid` file --> Get Info.
5. At the bottom, click on the lock icon and write your admin user and password and enable permission to your own user:
6. Click on the `+` button if you don't see your user name.

<img width="257" alt="SCR-20230309-fd0" src="https://user-images.githubusercontent.com/104392087/224082662-ef57ffda-c7e9-4314-bdae-510db167eb38.png">

> Note: I don't remember if I uninstalled `skhd` before doing this, so if it doesn't work you could try that first, and then doing this.

> Also, I had to enable a root password, because it wouldn't let me change permissions not even using the `chmod -R 755` command. 

You can see here the `/private/` directory with all of its permissions enabled bc I tested this command on the terminal after enabling `root password`.

<img width="787" alt="SCR-20230309-f9a" src="https://user-images.githubusercontent.com/104392087/224081447-a14092c9-19e9-4d75-ae5d-34989df2d995.png">

And here are the permissions for my `/private/tmp/` directory in which I had enabled only **write and read** user permissions. 

<img width="1203" alt="SCR-20230309-fgs" src="https://user-images.githubusercontent.com/104392087/224083581-f40f937c-b34c-4872-86ee-2b307a531dbc.png">
